### ISSUE TITLE
Error in yarn run gulp

### ISSUE THREAD

**Issue by adsazad** [Sylius/Sylius#11760]
**Sylius version affected**: 1.7.0

**Description**  
```
$ gulp build
[17:50:03] Requiring external module babel-register
[17:50:05] Using gulpfile ~/public_html/gulpfile.babel.js
[17:50:05] Starting 'build'...
[17:50:05] Starting 'buildAdmin'...
[17:50:05] Starting 'buildShop'...
[gulp-chug] Gulpfile, gulpfile.babel.js, contents is empty. Reading directly from disk...
[gulp-chug] Spawning process /home/syilusdemo/public_html/node_modules/gulp/bin/gulp.js with args /home/syilusdemo/public_html/node_modules/gulp/bin/gulp.js --gulpfile gulpfile.babel.js build --rootPath ../../../../../../../public/assets --nodeModulesPath ../../../../../../../node_modules from directory /home/syilusdemo/public_html/vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle...
[gulp-chug] Gulpfile, gulpfile.babel.js, contents is empty. Reading directly from disk...
[gulp-chug] Spawning process /home/syilusdemo/public_html/node_modules/gulp/bin/gulp.js with args /home/syilusdemo/public_html/node_modules/gulp/bin/gulp.js --gulpfile gulpfile.babel.js build --rootPath ../../../../../../../public/assets --nodeModulesPath ../../../../../../../node_modules from directory /home/syilusdemo/public_html/vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle...
[gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) node[106]: ../src/node_platform.cc:62:std::unique_ptr<long unsigned int> node::WorkerThreadsTaskRunner::DelayedTaskScheduler::Start(): Assertion `(0) == (uv_thread_create(t.get(), start_thread, this))' failed.
[gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js)  1: 0xa093f0 node::Abort() [node]
 2: 0xa0946e  [node]
 3: 0xa7905a node::WorkerThreadsTaskRunner::WorkerThreadsTaskRunner(int) [node]
 4: 0xa7911b node::NodePlatform::NodePlatform(int, node::tracing::TracingController*) [node]
[gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js)  5: 0x9da10b node::InitializeOncePerProcess(int, char**) [node]
[gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js)  6: 0x9da333 node::Start(int, char**) [node]
 7: 0x7ff6489df555 __libc_start_main [/lib64/libc.so.6]
[gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js)  8: 0x979215  [node]
[17:50:05] 'buildShop' errored after 107 ms
[17:50:05] Error in plugin 'gulp-chug'
Message:
    Gulpfile /home/syilusdemo/public_html/vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js exited with an error :(
Details:
    domainEmitter: [object Object]
    domainThrown: false
[17:50:05] 'build' errored after 112 ms
error Command failed with exit code 1.
```

**Steps to reproduce**  
i am running it on my server.
nodejs: v12.18.3
yarn: 1.22.4

**Comment by adsazad** [comment]
Still nothing

**Comment by thijskuilman** [comment]
I got the same problem:

nodejs: v10.19.0
yarn: 1.22.5

**Comment by lchrusciel** [comment]
Hey folks,

sorry for the late response, but the reported bug is hard to be reproduced. It should work on:
nodejs: v14.14.0
yarn: 1.22.10

**Comment by tper83** [comment]
Hi, for me it does not work on whatever node/yarn version I checked. 

Solution for this was was:
1) remove /node_modules
2) npm install first
3) yarn build
4) yarn encore prod

Don't know why, but creating /node_modules by 'yarn build' was breaking it.

**Comment by wadjeroudi** [comment]
maybe it can help, had the problem when our server was in a low memory available state.

**Comment by danielkeho** [comment]
It seems  that it's a permission problem.I'm on linux ubuntu and in root mode it works by doing **_sudo gulp build_**

**Comment by xchopin** [comment]
same issue on macos

**Comment by adsazad** [comment]
No we can't use sudo on server.

**Comment by Gilbertelnino** [comment]
If you are using xampp, please stop its server and then run  `yarn build`  or `npm run build` again. it will work

**Comment by adsazad** [comment]
Not Xampp. its CPANEL

**Comment by Zyten** [comment]
Just as an update, this still happens in Sylius 1.9.0

Environment:
Node v14.16.1
NPM: 6.14.12

**Comment by giosh94mhz** [comment]
This happen to me a lot today, and in the end was due to memory exausted. I've configured a small web server (for a very small ecommerce) which actually has only 1CPU, 1GB RAM and 1GB swapfile; once the error memory fills up, this error shows up.

I managed to workaround this issue by stopping some service during build (php-fpm), and then restarting once the compilation completes.

I think a viable solution could be to remove/reduce gulp parallelism in case of small servers.
